src/Log/Psr3Logger.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace ShoppinPal\YapepCommon\Log;
+
+use Psr\Log\LoggerInterface;
+use YapepBase\Config;
+use YapepBase\Log\LoggerAbstract;
+use YapepBase\Log\Message\IMessage;
+
+class Psr3Logger extends LoggerAbstract
+{
+
+    const APPLICATION_UNKNOWN = 'UNKNOWN';
+
+    /** @var string */
+    protected $programName;
+
+    /** @var string */
+    protected $applicationName;
+
+    /** @var LoggerInterface */
+    protected $logger;
+
+    /**
+     * @param LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+
+        $this->programName = Config::getInstance()->get('system.project.name');
+    }
+
+    /**
+     * @return LoggerInterface
+     */
+    public function getPsr3Logger()
+    {
+        return $this->logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function logMessage(IMessage $message)
+    {
+        $context = [
+            'program'     => $this->programName,
+            'application' => $this->getApplicationName(),
+            'hostname'    => php_uname('n'),
+            'language'    => 'php',
+            'pid'         => getmypid(),
+            'sapi'        => PHP_SAPI,
+            'tag'         => $message->getTag(),
+            'fields'      => $message->getFields(),
+        ];
+
+
+        switch ($message->getPriority()) {
+            case LOG_DEBUG:
+                $this->logger->debug($message->getMessage(), $context);
+                break;
+
+            case LOG_INFO:
+                $this->logger->info($message->getMessage(), $context);
+                break;
+
+            case LOG_NOTICE:
+                $this->logger->notice($message->getMessage(), $context);
+                break;
+
+            case LOG_WARNING:
+                $this->logger->warning($message->getMessage(), $context);
+                break;
+
+            case LOG_ERR:
+                $this->logger->error($message->getMessage(), $context);
+                break;
+
+            case LOG_CRIT:
+                $this->logger->critical($message->getMessage(), $context);
+                break;
+
+            case LOG_ALERT:
+                $this->logger->alert($message->getMessage(), $context);
+                break;
+
+            case LOG_EMERG:
+                $this->logger->emergency($message->getMessage(), $context);
+                break;
+        }
+    }
+
+    /**
+     * @return string
+     */
+    protected function getApplicationName()
+    {
+        if (null !== $this->applicationName) {
+            return $this->applicationName;
+        }
+
+        $applicationName = Config::getInstance()->get('application.name', self::APPLICATION_UNKNOWN);
```

Comment:
Is this needed to be able to log when the application is not set up yet?

Reply:
Yes, if something goes wrong during the basic bootstrapping process before we have the application name set, then we should still be able to log it, though without the application name. Another possibility is if a project/application is incorrectly configured and simply doesn't have this config value set.